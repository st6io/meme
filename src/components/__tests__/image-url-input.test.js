import React from 'react';
import { shallow } from 'enzyme';

import ImageUrlInput from '../image-url-input';
import { Input } from '../';

describe('<ImageUrlInput />', () => {
  const getComponent = props =>
    shallow(<ImageUrlInput onChange={jest.fn()} {...props} />);

  it('should match snapshot when Input not visible', () => {
    expect(getComponent()).toMatchSnapshot();
  });

  it('should match snapshot when Input should be visible', () => {
    expect(
      getComponent({ visible: true, value: 'http://test.url.com' }),
    ).toMatchSnapshot();
  });

  it('should invoke onChange Prop', () => {
    const onChangeMock = jest.fn();
    const component = getComponent({
      visible: true,
      onChange: onChangeMock,
    });

    expect(onChangeMock).toHaveBeenCalledTimes(0);
    const input = component.find(Input);
    const value = 'https://test.test';
    input.simulate('change', value);

    expect(onChangeMock).toHaveBeenCalledTimes(1);
    expect(onChangeMock).toHaveBeenCalledWith(value);
  });
});
