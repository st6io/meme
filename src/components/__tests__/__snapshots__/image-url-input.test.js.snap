// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageUrlInput /> should match snapshot when Input not visible 1`] = `
<CSSTransition
  classNames="image-url"
  in={false}
  mountOnEnter={true}
  timeout={
    Object {
      "enter": 300,
      "exit": 400,
    }
  }
  unmountOnExit={true}
>
  <TransitionContainer
    flexDirection="column"
    key="image-url-input-container"
    px={3}
  >
    <Styled(Styled(styled.div))
      as="input"
      border="1px solid"
      borderRadius={3}
      fontSize={1}
      mb={3}
      mx={0}
      onChange={[MockFunction]}
      placeholder="Image URL..."
      px={3}
      py={2}
      variant="primary"
    />
  </TransitionContainer>
</CSSTransition>
`;

exports[`<ImageUrlInput /> should match snapshot when Input should be visible 1`] = `
<CSSTransition
  classNames="image-url"
  in={true}
  mountOnEnter={true}
  timeout={
    Object {
      "enter": 300,
      "exit": 400,
    }
  }
  unmountOnExit={true}
>
  <TransitionContainer
    flexDirection="column"
    key="image-url-input-container"
    px={3}
  >
    <Styled(Styled(styled.div))
      as="input"
      border="1px solid"
      borderRadius={3}
      fontSize={1}
      mb={3}
      mx={0}
      onChange={[MockFunction]}
      placeholder="Image URL..."
      px={3}
      py={2}
      value="http://test.url.com"
      variant="primary"
    />
  </TransitionContainer>
</CSSTransition>
`;
